src/NumericalAlgorithms/DiscontinuousGalerkin/MetricIdentityJacobian.cpp

```diff
@@ -232,28 +234,146 @@ void metric_identity_jacobian(
   }
 }
 
-template void metric_identity_jacobian(
-    gsl::not_null<
-        InverseJacobian<DataVector, 1, Frame::Logical, Frame::Inertial>*>
-        det_jac_times_inverse_jacobian,
-    const Mesh<1>& mesh,
-    const tnsr::I<DataVector, 1, Frame::Inertial>& inertial_coords,
-    const Jacobian<DataVector, 1, Frame::Logical, Frame::Inertial>&
-        jacobian) noexcept;
-template void metric_identity_jacobian(
-    gsl::not_null<
-        InverseJacobian<DataVector, 2, Frame::Logical, Frame::Inertial>*>
-        det_jac_times_inverse_jacobian,
-    const Mesh<2>& mesh,
-    const tnsr::I<DataVector, 2, Frame::Inertial>& inertial_coords,
-    const Jacobian<DataVector, 2, Frame::Logical, Frame::Inertial>&
-        jacobian) noexcept;
-template void metric_identity_jacobian(
-    gsl::not_null<
-        InverseJacobian<DataVector, 3, Frame::Logical, Frame::Inertial>*>
+template <size_t Dim>
+void metric_identity_jacobian_and_inv_jacobian(
+    const gsl::not_null<
+        InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Inertial>*>
         det_jac_times_inverse_jacobian,
-    const Mesh<3>& mesh,
-    const tnsr::I<DataVector, 3, Frame::Inertial>& inertial_coords,
-    const Jacobian<DataVector, 3, Frame::Logical, Frame::Inertial>&
-        jacobian) noexcept;
+    const gsl::not_null<
+        InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Inertial>*>
+        inverse_jacobian,
+    const gsl::not_null<
+        Jacobian<DataVector, Dim, Frame::Logical, Frame::Inertial>*>
+        jacobian,
+    const gsl::not_null<Scalar<DataVector>*> det_jacobian,
+    const Mesh<Dim>& mesh,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& inertial_coords) noexcept {
+  static_assert(Dim == 1 or Dim == 2 or Dim == 3,
+                "Only implemented for 1, 2, and 3d.");
+  destructive_resize_components(det_jac_times_inverse_jacobian,
+                                mesh.number_of_grid_points());
+  destructive_resize_components(inverse_jacobian, mesh.number_of_grid_points());
+  destructive_resize_components(det_jacobian, mesh.number_of_grid_points());
+  // ASSERT();
```

Comment:
Remove... unless if maybe it'd be useful to ASSERT that the input/input `jacobian` also matches the mesh size?